CHANGELOG.md

```diff
@@ -2,6 +2,12 @@
 
 #### Upcoming Changes
 
+* Fix `CairoRunner::get_memory_holes`
+
+    * Skip builtin segements when counting memory holes
+    * Check amount of memory holes for all tests in cairo_run_test
+    * Remove duplicated tests in cairo_run_test
+
```

Comment:
This is a breaking change AFAICT. `count_memory_holes` is marked as public and its signature changed.

Reply:
Oh, I thought only `CairoRuner::get_memory_holes` was public, maybe we should consider making this one private, as the `MemorySegmentManager` is a private field of the VM. Users should only really  be using``CairoRunner::get_memory_holes`. What do you think?

Reply:
I'm all for privacy. It's not breaking if there's no way to call it tho, but it's better to be explicit. Is the structure itself private?

Reply:
It would still be a breaking change (making something private), The structure itself is not private (`MemorySegmentManager`), but now you would need to get `vm.builtin_runners.len()` (another private field) to use it effectively, so there is not much reason to keep it in the public interface imo.

Reply:
Yes, it would. I'm not arguing for keeping it public, I'm just asking whether or not it really breaks to document properly. But I'm in favor of making both the structure and the method private.

Reply:
Structure is still useful for the `VirtualMachineBuilder`, so Ill just make the method private

Reply:
I'm not sure what you mean but I don't want to waste your time and we can go back to that later. Let's leave it on making the method private and counting it as a breaking change just in case, do you agree?